src/gui/preferences.c

```diff
@@ -706,44 +706,72 @@ static void tree_insert_presets(GtkTreeStore *tree_model)
     const gboolean writeprotect = (sqlite3_column_int(stmt, 15) == 0 ? FALSE : TRUE);
 
-    gchar *iso = NULL, *exposure = NULL, *aperture = NULL, *focal_length = NULL;
+    gchar *iso = NULL, *exposure = NULL, *aperture = NULL, *focal_length = NULL, *smaker = NULL, *smodel = NULL, *slens = NULL;
     int min, max;
 
     gchar *module = g_strdup(dt_iop_get_localized_name(operation));
     if(module == NULL) module = g_strdup(dt_lib_get_localized_name(operation));
     if(module == NULL) module = g_strdup(operation);
 
-    if(iso_min == 0.0 && iso_max == FLT_MAX)
-      iso = g_strdup("%");
-    else
-      iso = g_strdup_printf("%zu – %zu", (size_t)iso_min, (size_t)iso_max);
+    gboolean islib = FALSE;
 
-    min = 0, max = 0;
-    for(; min < dt_gui_presets_exposure_value_cnt && exposure_min > dt_gui_presets_exposure_value[min]; min++)
-      ;
-    for(; max < dt_gui_presets_exposure_value_cnt && exposure_max > dt_gui_presets_exposure_value[max]; max++)
-      ;
-    if(min == 0 && max == dt_gui_presets_exposure_value_cnt - 1)
-      exposure = g_strdup("%");
-    else
-      exposure = g_strdup_printf("%s – %s", dt_gui_presets_exposure_value_str[min],
-                                 dt_gui_presets_exposure_value_str[max]);
+    for (const GList * lib_modules = g_list_first(darktable.lib->plugins); lib_modules; lib_modules = g_list_next(lib_modules))
+    {
+      dt_lib_module_t *lib_module = (dt_lib_module_t *)lib_modules->data;
+      if(!strcmp(lib_module->plugin_name, operation))
+      {
+        islib = TRUE;
+      }
+    }
 
-    min = 0, max = 0;
-    for(; min < dt_gui_presets_aperture_value_cnt && aperture_min > dt_gui_presets_aperture_value[min]; min++)
-      ;
-    for(; max < dt_gui_presets_aperture_value_cnt && aperture_max > dt_gui_presets_aperture_value[max]; max++)
-      ;
-    if(min == 0 && max == dt_gui_presets_aperture_value_cnt - 1)
-      aperture = g_strdup("%");
+    if(islib)
+    {
+      iso = g_strdup("-");
+      exposure = g_strdup("-");
+      aperture = g_strdup("-");
+      focal_length = g_strdup("-");
+      smaker = g_strdup("-");
+      smodel = g_strdup("-");
+      slens = g_strdup("-");
+    }
     else
-      aperture = g_strdup_printf("%s – %s", dt_gui_presets_aperture_value_str[min],
-                                 dt_gui_presets_aperture_value_str[max]);
+    {
+      smaker = g_strdup(maker);
+      smodel = g_strdup(model);
+      slens = g_strdup(lens);
 
-    if(focal_length_min == 0.0 && focal_length_max == 1000.0)
-      focal_length = g_strdup("%");
-    else
-      focal_length = g_strdup_printf("%d – %d", focal_length_min, focal_length_max);
+      if(iso_min == 0.0 && iso_max == FLT_MAX)
+        iso = g_strdup("%");
+      else
+        iso = g_strdup_printf("%zu – %zu", (size_t)iso_min, (size_t)iso_max);
+
+      min = 0, max = 0;
+      for(; min < dt_gui_presets_exposure_value_cnt && exposure_min > dt_gui_presets_exposure_value[min]; min++)
```

Comment:
I know this is not your code, but I would put `min = 0` within the `for(`; it looks funny this way.